pandas/core/computation/expr.py

```diff
@@ -702,7 +702,11 @@ def visit_Call(self, node, side=None, **kwargs):
                 if key.arg:
                     kwargs[key.arg] = self.visit(key.value).value
 
-            return self.const_type(res(*new_args, **kwargs), self.env)
+            if self.engine == "numexpr":
```

Comment:
can we just *always* do this?

Reply:
Indeed, This was one of my previous comments. It could depend on the type of the result returned by the function. But since the result has to be embedded in an expression string passed to `numexpr.evaluate()` it's probably safer to use variable names instead of string representations.

Unless it has an interest for some very specific cases like some literals of base types.

Reply:
ok if you can try that would be great

Reply:
I've done some tests with numexpr.

For instance this works:
```python
numexpr.evaluate("2 * a + b", {"a": np.array([0.0, 0.0]), "b": [1, 2]})
```
while this doesn't:
```python
numexpr.evaluate("2 * a + [1, 2]", {"a": np.array([0.0, 0.0])})
```
So I guess the number of cases where it's a bad idea to use variables instead of literals is quite reduced.
It would basically work only for integer literals basically (doing it for floats would lead to truncation). Is it worth making such an exception ?

Reply:
no, try to make the change to just fix this (of course if you can additional tests would be great)

Reply:
no what i mean is that i would like to have your patch handle all of the cases and not special case (which i think it works).